### ISSUE TITLE
Build: introduce "keep going" concept for build commands

### ISSUE THREAD

**Issue by humitos** [readthedocs/readthedocs.org#9137]
Currently, when a command finishes with non-0 exit code Read the Docs stops and fails the build immediately. This is a feature in itself because the build requires all the steps to success to be able to publish the documentation, and failing it fast improves UX.

However, there are some scenarios, in particular now with `build.jobs`, where being able to run a command immediately after another command that has failed could be useful. We just had a good example where the ability to do `cat .../report/*.log` would have helped to debug a build that was failing _only_ on Rea the Docs, but not locally. See https://github.com/executablebooks/jupyter-book/pull/1711

Basically, what I'm proposing is to think of some other scenarios where this feature could be useful. The feature would be like this:

```yaml
build:
  keep_going: true
```

and it will make Read the Docs execute _all_ the commands; continuing even if any of them return non-0 exit code. However, it will fail the build if any command returns a non-0 exit code and won't upload the resulting documentation built.

> Note that Sphinx already has this concept itself and we support it via `sphinx.fail_on_warnings: true` (see https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-keep-going)

**Comment by stsewd** [comment]
GitHub actions have https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error for each command, there is also the old trick of executing the command like `cmd || true`.

But I think what would be more useful is to mark the command in red or yellow, if we mark it as another command it won't clear to the user that the command failed.

**Comment by humitos** [comment]
@stsewd 

> there is also the old trick of executing the command like `cmd || true`.

This works when using `build.jobs` to write custom commands. However, it's not possible for default commands executed by Read the Docs in all the other steps. I'm interested in failing those commands (e.g. `sphinx-build`) and keeping going.

**Comment by agjohnson** [comment]
> there is also the old trick of executing the command like cmd || true

I think users would do this in most cases.

> I'm interested in failing those commands (e.g. sphinx-build) and keeping going

This is an interesting idea, I see a few use cases for this feature.

I'd rather keep this all using the same implementation though. I'd rather see a job that was able to execute at the end of the process, regardless of build errors -- `build.jobs.final`, `build.jobs.on_error`, `build.jobs.post_cleanup`, or similar.
